pkg/v2/crud/traverse.go

```diff
@@ -2,21 +2,56 @@ package crud
 
 import (
 	"fmt"
+
 	"github.com/imulab/go-scim/pkg/v2/annotation"
 	"github.com/imulab/go-scim/pkg/v2/crud/expr"
 	"github.com/imulab/go-scim/pkg/v2/prop"
 	"github.com/imulab/go-scim/pkg/v2/spec"
 )
 
-func defaultTraverse(property prop.Property, query *expr.Expression, callback func(nav prop.Navigator) error) error {
-	return traverser{
+type traverseCb func(nav prop.Navigator) error
+type traverseUpdatedValueCb func(nav prop.Navigator, value interface{}) error
```

Comment:
is this consistent with other type definitions in the library?

Reply:
it's not very popular among the lib to define a function type:
```
grep -rn type | grep func

./prop/subscriber.go:41:type SubscriberFactoryFunc func(publisher Property, params map[string]interface{}) Subscriber
./spec/attribute.go:162:// ForEachReferenceTypes invokes callback function on each defined reference types
./facade/support.go:46:func typeCheck(attr *spec.Attribute, t reflect.Type) error {
./crud/traverse.go:12:type traverseCb func(nav prop.Navigator) error
./crud/traverse.go:13:type traverseUpdatedValueCb func(nav prop.Navigator, value interface{}) error
./crud/traverse.go:187:type elementStrategy func(multiValuedComplex prop.Property) func(index int, child prop.Property) bool
```